+ source ../common/common_lib.sh
++ source /root/GitRepo/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/GitRepo/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Fri Jan  6 03:11:52 2023 - INFO  - Start environmental preparation.
+ DNF_INSTALL edac-utils
+ pkgs=edac-utils
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/GitRepo/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs edac-utils --node 1 --tempfile ''
+ tmpfile2=/tmp/tmpl9qemn6m
+ '[' -z '' ']'
+ tmpfile=/tmp/tmpl9qemn6m
+ service=edac.service
++ date '+%Y-%m-%d %T'
+ log_time='2023-01-06 03:13:12'
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Fri Jan  6 03:13:13 2023 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Fri Jan  6 03:13:15 2023 - INFO  - Start testing...
+ systemctl restart edac.service
Job for edac.service failed because the control process exited with error code.
See "systemctl status edac.service" and "journalctl -xeu edac.service" for details.
+ CHECK_RESULT 1 0 0 'edac.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='edac.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'edac.service restart failed'
+ LOG_ERROR 'edac.service restart failed'
+ message='edac.service restart failed'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'edac.service restart failed'
Fri Jan  6 03:13:18 2023 - ERROR - edac.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_edac.sh line 33'
+ message='oe_test_service_edac.sh line 33'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_edac.sh line 33'
Fri Jan  6 03:13:20 2023 - ERROR - oe_test_service_edac.sh line 33
+ return 0
+ systemctl stop edac.service
+ CHECK_RESULT 0 0 0 'edac.service stop failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='edac.service stop failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl start edac.service
Job for edac.service failed because the control process exited with error code.
See "systemctl status edac.service" and "journalctl -xeu edac.service" for details.
+ CHECK_RESULT 1 0 0 'edac.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='edac.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'edac.service start failed'
+ LOG_ERROR 'edac.service start failed'
+ message='edac.service start failed'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'edac.service start failed'
Fri Jan  6 03:13:23 2023 - ERROR - edac.service start failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_edac.sh line 37'
+ message='oe_test_service_edac.sh line 37'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_edac.sh line 37'
Fri Jan  6 03:13:24 2023 - ERROR - oe_test_service_edac.sh line 37
+ return 0
+ grep 'Active: active'
+ systemctl status edac.service
+ CHECK_RESULT 1 0 0 'edac.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='edac.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'edac.service start failed'
+ LOG_ERROR 'edac.service start failed'
+ message='edac.service start failed'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'edac.service start failed'
Fri Jan  6 03:13:26 2023 - ERROR - edac.service start failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_edac.sh line 39'
+ message='oe_test_service_edac.sh line 39'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_edac.sh line 39'
Fri Jan  6 03:13:27 2023 - ERROR - oe_test_service_edac.sh line 39
+ return 0
+ test_enabled edac.service
+ service=edac.service
++ systemctl is-enabled edac.service
+ state=disabled
+ '[' disabled == enabled ']'
+ '[' disabled == disabled ']'
++ systemctl enable edac.service
++ grep 'Created symlink'
++ awk '{print $3}'
+ symlink_file=/etc/systemd/system/multi-user.target.wants/edac.service
+ find /etc/systemd/system/multi-user.target.wants/edac.service
/etc/systemd/system/multi-user.target.wants/edac.service
+ CHECK_RESULT 0 0 0 'edac.service enable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='edac.service enable failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl disable edac.service
Removed /etc/systemd/system/multi-user.target.wants/edac.service.
+ find /etc/systemd/system/multi-user.target.wants/edac.service
find: ‘/etc/systemd/system/multi-user.target.wants/edac.service’: No such file or directory
+ CHECK_RESULT 1 0 1 'edac.service disable failed'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='edac.service disable failed'
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ journalctl --since '2023-01-06 03:13:12' -u edac.service
+ grep -i 'fail\|error'
+ grep -v -i 'DEBUG\|INFO\|WARNING'
+ grep -v -i 'Error: No dimm labels for'
Jan 06 03:13:17 openEuler-riscv64 edac-ctl[2054]: failed to run : Broken pipe
Jan 06 03:13:17 openEuler-riscv64 systemd[1]: edac.service: Failed with result 'exit-code'.
Jan 06 03:13:17 openEuler-riscv64 systemd[1]: Failed to start Initialize EDAC Drivers For Machine Hardware.
Jan 06 03:13:21 openEuler-riscv64 edac-ctl[2531]: failed to run : Broken pipe
Jan 06 03:13:21 openEuler-riscv64 systemd[1]: edac.service: Failed with result 'exit-code'.
Jan 06 03:13:21 openEuler-riscv64 systemd[1]: Failed to start Initialize EDAC Drivers For Machine Hardware.
+ CHECK_RESULT 0 1 0 'There is an error message for the log of edac.service'
+ actual_result=0
+ expect_result=1
+ mode=0
+ error_log='There is an error message for the log of edac.service'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 1x
+ test -n 'There is an error message for the log of edac.service'
+ LOG_ERROR 'There is an error message for the log of edac.service'
+ message='There is an error message for the log of edac.service'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'There is an error message for the log of edac.service'
Fri Jan  6 03:13:43 2023 - ERROR - There is an error message for the log of edac.service
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_edac.sh line 42'
+ message='oe_test_service_edac.sh line 42'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_edac.sh line 42'
Fri Jan  6 03:13:44 2023 - ERROR - oe_test_service_edac.sh line 42
+ return 0
+ service edac.service reload
+ grep 'Job type reload is not applicable'
+ CHECK_RESULT 1 0 0 'Job type reload is not applicable for unit edac.service'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Job type reload is not applicable for unit edac.service'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Job type reload is not applicable for unit edac.service'
+ LOG_ERROR 'Job type reload is not applicable for unit edac.service'
+ message='Job type reload is not applicable for unit edac.service'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'Job type reload is not applicable for unit edac.service'
Fri Jan  6 03:13:45 2023 - ERROR - Job type reload is not applicable for unit edac.service
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_edac.sh line 44'
+ message='oe_test_service_edac.sh line 44'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_edac.sh line 44'
Fri Jan  6 03:13:46 2023 - ERROR - oe_test_service_edac.sh line 44
+ return 0
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Fri Jan  6 03:13:47 2023 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 5
+ test 5 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Fri Jan  6 03:13:48 2023 - ERROR - failed to execute the case.
+ exit 5
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Fri Jan  6 03:13:49 2023 - INFO  - start environment cleanup.
+ systemctl stop edac.service
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmpl9qemn6m ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmpl9qemn6m
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/GitRepo/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Fri Jan  6 03:14:13 2023 - INFO  - Finish environment cleanup!
